src/db/oracledb/jobs-dao.js

```diff
@@ -37,23 +120,316 @@ const getJobs = async (internalId, query) => {
+
+  const result = await connection.execute(contrib.studentUpdateJob(binds), binds);
+  return result.outBinds.result;
+};
+
+/**
+ * Creates student job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const studentCreateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, studentBinds);
+
+  const { outBinds: { result } } = await connection.execute(contrib.studentCreateJob(binds), binds);
+  return result;
+};
+
+/**
+ * Creates graduate job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const graduateCreateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib.graduateCreateJob(binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Updates graduate job records
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @returns {string} Query result, null if success
+ */
+const graduateUpdateJob = async (connection, osuId, body) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib.graduateUpdateJob(binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Determines if changeReasonCode is valid
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} changeReasonCode Change reason code from request body
+ * @returns {boolean} True if changeReasonCode is valid
+ */
+const isValidChangeReasonCode = async (connection, changeReasonCode) => {
+  const { rows } = await connection.execute(
+    contrib.validateChangeReasonCode(),
+    { changeReasonCode },
+  );
+  return rows[0].count > 0;
+};
+
+/**
+ * Determines which Epaf to execute based on fields in body
+ *
+ * @param {boolean} update True if updating record. False if creating record
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @param {string} internalId Internal ID of a person
+ * @returns {object} raw job record
+ */
+const createOrUpdateJob = async (update, osuId, body) => {
+  const connection = await getConnection('banner');
+  try {
+    let result;
+    const { studentEmployeeInd, changeReason: { code: changeReasonCode } } = body;
+    const employmentType = studentEmployeeInd ? 'student' : 'graduate';
 
-    if (rows.length > 1) {
-      throw new Error(`Multiple job records found for job ID ${jobId}`);
+    if (employmentType === 'student' && !/^C5[0-2]/.test(body.positionNumber)) {
+      return new Error('Position number for students must start with C50, C51, or C52');
```

Comment:
Interesting, I don't think we have this in v1?

Reply:
It is probably a 500 error then. I got the error text from the oracledb error message.